src/docfx/build/metadata/Metadata.cs

```diff
@@ -21,5 +28,101 @@ where item.Match(file.FilePath)
 
             return JsonUtility.Merge(config.GlobalMetadata, fileMetadata);
         }
+
+        public static JObject GenerateRawMetadata(Context context, Document file, TableOfContentsMap tocMap, string html)
+        {
+            var rawMetadata = new JObject();
+            var locale = file.Docset.Config.Locale;
+            var depotName = $"{file.Docset.Config.Product}{file.Docset.Config.DocsetName}";
+
+            rawMetadata["_op_canonicalUrlPrefix"] = $"https://docs.microsoft.com/{file.Docset.Config.Locale}/{file.Docset.Config.SiteBasePath}";
+            rawMetadata["_op_pdfUrlPrefixTemplate"] = $"https://docs.microsoft.com/pdfstore/{locale}/{depotName}/{{branchName}}{{pdfName}}";
+            rawMetadata["_op_rawTitle"] = GetRawTitle(html);
+
+            rawMetadata["_op_wordCount"] = rawMetadata["word_count"] = CountWord(html);
+
+            rawMetadata["depot_name"] = depotName;
+            rawMetadata["is_dynamic_rendering"] = true;
+            rawMetadata["layout"] = "Conceptual";
+            rawMetadata["locale"] = locale;
+
+            rawMetadata["site_name"] = "Docs";
+            rawMetadata["toc_rel"] = tocMap.FindTocRelativePath(file);
+            rawMetadata["version"] = 0;
+
+            return rawMetadata;
+        }
+
+        internal static long CountWord(string html)
+        {
+            // TODO: word count does not work for CJK locales...
+            if (html == null)
+            {
+                throw new ArgumentNullException(nameof(html));
+            }
+
+            HtmlDocument document = new HtmlDocument();
+
+            // Append a space before each end bracket so that InnerText inside different child nodes can separate itself from each other.
+            document.LoadHtml(html.Replace("</", " </", StringComparison.OrdinalIgnoreCase));
```

Comment:
we should only load html once.

Reply:
in word counting, we have to do some replacement to the html, which can not be shared with others.

---
In reply to: [192974761](https://github.com/dotnet/docfx/pull/2840#discussion_r192974761) [](ancestors = 192974761)